### ISSUE TITLE
Timber 2.0 Twig Cache not working on render method

### ISSUE THREAD

**Issue by billybigpotatoes** [timber/timber#2429]
Using render method on single-post template - all posts are being cached - context appears to be ignored

## Expected behavior

I would expect to see different posts based on the post slug

## Actual behavior

All posts are showing the same content ignoring the slug / context. The item displayed seems to be the first item that was requested once the cache was enabled

## Steps to reproduce behavior

Following page controller:  single-post.php 

```
<?php
require_once( __DIR__ . '/vendor/autoload.php' );
$timber = new Timber\Timber();
$start = Timber\Helper::start_timer();

$context = Timber::context();
// $context['time'] = date();
// echo "<pre>";
// echo print_r($context, true);
// echo "</pre>";

// Render Page
get_header();
Timber::render( 'presenter.twig', $context, 60 );
get_footer();
echo Timber\Helper::stop_timer( $start);
```

Issue is resolved by setting cache from 60 to false - but at the expense of cache

## What version of WordPress, PHP and Timber are you using?

Not sure if this is related to the deprecated cache lib?

- WP 5.7. 
- PHP 7.4.1

- altorouter/altorouter        2.0.1           A lightning fast router for PHP
- composer/installers          v1.10.0         A multi-framework Composer library installer
- palmiak/timber-acf-wp-blocks 1.14            Create Gutenberg blocks from Twig templates and ACF fields.
- symfony/polyfill-ctype       v1.22.1         Symfony polyfill for ctype functions
- symfony/polyfill-mbstring    v1.22.1         Symfony polyfill for the Mbstring extension
- timber/timber                2.x-dev 2509b5c Plugin to write WordPress themes w Object-Oriented Code and the Twig Template Engine
- twig/cache-extension         v1.5.0          Cache fragments of templates directly within Twig.
- twig/twig                    v2.14.4         Twig, the flexible, fast, and secure template language for PHP
- upstatement/routes           0.8             Manage rewrites and routes in WordPress with this dead-simple plugin

## How did you install Timber? (for example, from GitHub, Composer/Packagist, WP.org?)

Composer

**Comment by gchtr** [comment]
Hello @billybigpotatoes 

I’m trying to reproduce this issue and I think it would really help if you could also provide your Twig file. Are you looping over the posts in there?

In your example, I don’t see where your posts are coming from. In a singular template like **single-post.php**, there’s usually only one post to be displayed.

**Comment by gchtr** [comment]
Found out what the problem is. We’re going to keep track of this issue in #2646.

**Comment by gchtr** [comment]
Fixed in #2675. This will be in the next 2.x release (hopefully soon).
